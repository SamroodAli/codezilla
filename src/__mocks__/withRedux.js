import reduxThunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import PropTypes from 'prop-types';
import data from '../../public/data/courses.json';

const initialState = {
  videos: [data],
  filters: {
    React: true,
    Angular: true,
    Vue: true,
    'Functional Programming': true,
    'Object Oriented Programming': true,
    Microservices: true,
    'JAM Stack': true,
  },
  theme: 'light',
  currentVideo: null,
};

const mockStore = configureStore([reduxThunk]);
export const store = mockStore(initialState);

const WithRedux = ({ children }) => (
  <Provider store={store}><StaticRouter>{children}</StaticRouter></Provider>
);

WithRedux.propTypes = {
  children: PropTypes.node.isRequired,
};

export default WithRedux;
