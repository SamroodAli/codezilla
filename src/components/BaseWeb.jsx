import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { Client as Styletron } from 'styletron-engine-atomic';
import { Provider as StyletronProvider, DebugEngine } from 'styletron-react';
import { BaseProvider, LightTheme, DarkTheme } from 'baseui';
import { Button } from 'baseui/button';
import useActions from '../hooks/useActions';

const debug = process.env.NODE_ENV === 'production' ? undefined : new DebugEngine();
const engine = new Styletron();

const THEME = {
  light: 'light',
  dark: 'dark',
};

const toggleTheme = (theme) => (theme === THEME.light ? THEME.dark : THEME.light);

const BaseWeb = ({ children }) => {
  const theme = useSelector(({ theme }) => theme);
  const { setTheme } = useActions();

  return (
    <StyletronProvider value={engine} debug={debug} debugAfterHydration>
      <BaseProvider theme={theme === THEME.light ? LightTheme : DarkTheme}>
        <Button
          onClick={() => setTheme(toggleTheme)}
        >
          Toggle light/dark theme!
        </Button>
        {children}
      </BaseProvider>
    </StyletronProvider>
  );
};

BaseWeb.propTypes = {
  children: PropTypes.node.isRequired,
};

export default BaseWeb;
