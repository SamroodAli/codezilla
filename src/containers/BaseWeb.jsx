import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import useActions from '../hooks/useActions';
import BaseWeb from '../components/BaseWeb';

const BaseWebContainer = ({ children }) => {
  const theme = useSelector(({ theme }) => theme);
  const { setTheme } = useActions();

  return <BaseWeb theme={theme} setTheme={setTheme}>{ children}</BaseWeb>;
};
BaseWebContainer.propTypes = {
  children: PropTypes.node.isRequired,
};

export default BaseWebContainer;
