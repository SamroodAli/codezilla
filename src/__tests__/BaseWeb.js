import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import withRedux from 'withRedux';

import BaseWeb from '../components/BaseWeb';

describe('Testing BaseWeb with redux', () => {
  it('BaseWeb is defined', async () => {
    const baseweb = render(withRedux(<BaseWeb><p>Hello Base web</p></BaseWeb>));
    expect(baseweb).toBeDefined();
  });

  it('BaseWeb snapshot testing', () => {
    const tree = renderer.create(withRedux(<BaseWeb><p>Hello Base web</p></BaseWeb>)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
