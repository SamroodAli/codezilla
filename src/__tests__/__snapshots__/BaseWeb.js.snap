// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing BaseWeb with redux BaseWeb snapshot testing 1`] = `
Array [
  <div
    className=""
  >
    <nav
      className="ae af ag ah ai aj ak al am an ao"
      data-baseweb="header-navigation"
      role="navigation"
    >
      <ul
        className="ai ap aq ar as at au av aw ax ay az b0 b1 b2"
      >
        <li
          className="b3 au"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <div
              className="b4 ae b5 b6 b7"
              data-baseweb="typo-displaymedium"
            >
              CodeZilla
            </div>
          </a>
        </li>
      </ul>
      <ul
        className="ai ap aq b8 b9 ba au av bb bc ay az b0 b1 b2"
      />
      <ul
        className="ai ap aq ar as at au av bd be ay az b0 b1 b2"
      >
        <li
          className="b3 au"
        >
          <a
            href="/courses"
            onClick={[Function]}
          >
            <div
              className="b4 ae b5 bf ah"
              data-baseweb="typo-labellarge"
            >
              Courses
            </div>
          </a>
        </li>
      </ul>
      <ul
        className="ai ap aq ar as at au av bd be ay az b0 b1 b2"
      >
        <li
          className="b3 au"
        >
          <button
            className="bg bh bi bc bj bk bl bm bn bo bp bq br bs bt bu bv bw bx by bz c0 c1 b1 b0 b2 az ae af b5 c2 c3 c4 c5 c6 c7 c8 c9 ca cb cc cd ce"
            data-baseweb="button"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
          >
            Go dark
          </button>
        </li>
      </ul>
    </nav>
    <div
      className="cf cg"
      data-baseweb="block"
    >
      <p>
        Hello Base web
      </p>
    </div>
  </div>,
  <div
    className=""
  />,
]
`;
